src/libasr/containers.h

```diff
@@ -185,7 +185,7 @@ std::string string_format(const std::string& format, Args && ...args)
 {
     auto size = std::snprintf(nullptr, 0, format.c_str(), std::forward<Args>(args)...);
     std::string output(size, '\0');
-    std::sprintf(&output[0], format.c_str(), std::forward<Args>(args)...);
+    std::snprintf(&output[0], size + 1, format.c_str(), std::forward<Args>(args)...);
```

Comment:
@certik Is this change correct. I think size of the buffer should be ` + 1` because it wills only till second last position. Am I correct? Locally all the tests pass so seems like it. Let me know. I will merge this if you agree.

Reply:
Without ` + 1` I get incorrect updates in reference tests. So I think the above change is fine. Will merge once this test pass and no objections happen.